File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        v = set()
        tmp = headA
        while tmp:
            v.add(tmp)
            tmp = tmp.next
        tmp = headB
        while tmp and tmp not in v:
            tmp = tmp.next
        return tmp

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.java
import java.util.Set;
import java.util.HashSet;
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        Set<ListNode> v = new HashSet<ListNode>();
        ListNode tmp = headB;
        while(tmp != null){
            v.add(tmp);
            tmp = tmp.next;
        }
        tmp = headA;
        while(tmp != null && !v.contains(tmp)){
            tmp = tmp.next;
        }
        return (tmp != null) ? tmp : null;
    }
}